### PR TITLE
Make it easy to get a logger with context information

### PR DESCRIPTION
**Pull request by jml**
Related to https://github.com/weaveworks/service/issues/1233

### CODE REVIEW

File: logging/logging.go [hunk 1]
```diff
@@ -43,3 +46,21 @@ func (f *textFormatter) Format(entry *log.Entry) ([]byte, error) {
 	b.WriteByte('\n')
 	return b.Bytes(), nil
 }
+
+// WithContext returns a logger that has information from the context as fields.
+//
+// e.g.
+//     logger := logging.WithContext(ctx)
+//     logger.Errorf("Some error")
+func WithContext(ctx context.Context) *log.Entry {
```
Comment (tomwilkie) [inline logging/logging.go]: `WithUsedIDFromContext` might be more appropriate?  or `WithUsedID`?

Comment (jml) [inline logging/logging.go]: I figure if ever we add more information to the context, we might want to add more information here. `WithStandardWeaveworksContextInfo` is a bit wordy.

I guess instead we could add more `WithFoo` functions, since they do chain.

Comment (tomwilkie) [inline logging/logging.go]: How do they chain?

TBH I'd prefer if these lived in their various packages - ie this was in `users` as `Logger`, and there was one in the tracing package etc.  These could then be combined in a standard weaveworks logger here.  But just a thought.

**Comment by tomwilkie** [comment]
LGTM
